src/WebJobs.Script/Diagnostics/ValueStopwatch.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    // Based on: https://github.com/dotnet/aspnetcore/blob/master/src/Shared/ValueStopwatch/ValueStopwatch.cs
+    public struct ValueStopwatch
+    {
+        private static readonly double TimestampToTicks = TimeSpan.TicksPerSecond / (double)Stopwatch.Frequency;
+
+        private long _startTimestamp;
```

Comment:
Should it be `private readonly` ?

Reply:
`readonly` creates a defensive copy in the runtime, so that's why it's left off here - can add a comment to make clear!

Reply:
Comment added!